### PR TITLE
Eng 2021 implement notification repo

### PR DESCRIPTION
**Pull request by eunice-chan**
## Describe your changes and why you are making these changes
- Update src/golang/lib/models/shared/notification.go naming of constants to be clearer
- Implement sqlite + test cases

## Related issue number (if any)
Eng 2021

## Checklist before requesting a review
- [x] I have created a descriptive PR title. The PR title should complete the sentence "This PR...".
- [x] I have performed a self-review of my code.
- [N/A] I have included a small demo of the changes. For the UI, this would be a screenshot or a Loom video.
- [x] If this is a new feature, I have added unit tests and integration tests.
- [x] I have run the integration tests locally and they are passing.
- [x] I have run the linter script locally (See `python3 scripts/run_linters.py -h` for usage).
- [N/A] All features on the UI continue to work correctly.
- [x] Added one of the following CI labels:
    - `run_integration_test`: Runs integration tests
    - `skip_integration_test`: Skips integration tests (Should be used when changes are ONLY documentation/UI)

### CODE REVIEW

File: src/golang/lib/models/shared/notification.go [hunk 1]
```diff
@@ -10,26 +10,26 @@ import (
...
 
 type NotificationObject string
 
 const (
-	WorkflowObject     NotificationObject = "workflow"
-	DAGResultObject    NotificationObject = "workflow_dag_result"
-	OrganizationObject NotificationObject = "organization"
+	WorkflowNotificationObject     NotificationObject = "workflow"
+	DAGResultNotificationObject    NotificationObject = "workflow_dag_result"
+	OrganizationNotificationObject NotificationObject = "organization"
```
Comment (saurav-c) [inline src/golang/lib/models/shared/notification.go:32]: nit: we can shorten this to OrgNotificationObject

File: src/golang/lib/repos/tests/suite.go [hunk 2]
```diff
@@ -22,6 +22,7 @@ type TestSuite struct {
 	// List of all repos
 	workflow repos.Workflow
 	user     repos.User
+	notification repos.Notification
```
Comment (saurav-c) [inline src/golang/lib/repos/tests/suite.go:25]: nit: lets keep these is alphabetical order

File: src/golang/lib/repos/tests/suite.go [hunk 3]
```diff
@@ -41,6 +42,8 @@ func (ts *TestSuite) SetupSuite() {
 	// Initialize repos
 	ts.workflow = sqlite.NewWorklowRepo()
 	ts.user = sqlite.NewUserRepo()
+	ts.notification = sqlite.NewNotificationRepo()
```
Comment (saurav-c) [inline src/golang/lib/repos/tests/suite.go:45]: same nit as above

File: src/golang/lib/repos/tests/seed.go [hunk 4]
```diff
@@ -12,6 +12,27 @@ const (
 	testOrgID = "aqueduct-test"
 )
 
+// seedNotification creates count notification records.
+func (ts *TestSuite) seedNotification(count int) []models.Notification {
+	notifications := make([]models.Notification, 0, count)
+
+	for i := 0; i < count; i++ {
+		receiverID := uuid.New()
```
Comment (saurav-c) [inline src/golang/lib/repos/tests/seed.go:20]: I dont think sqlite will throw an error but technically the receiverID is a foreign key with the users table, so lets either require a user to be passed into this seed fn or generate a random user to use. Can you comment which approach you take as part of the function comment here

**Review by saurav-c — COMMENTED** [review]
I think needs to be rebased. Can you take a look why the unit tests are failing. Otherwise LGTM

File: src/golang/lib/execution_environment/dependency.go [hunk 5]
```diff
@@ -3,7 +3,7 @@ package execution_environment
 import (
 	"archive/zip"
 	"bytes"
-	"io/ioutil"
```
Comment (eunice-chan) [inline src/golang/lib/execution_environment/dependency.go:6]: Linter complaint:
```
lib/execution_environment/dependency.go:6:2: SA1019: "io/ioutil" has been deprecated since Go 1.16: As of Go 1.16, the same functionality is now provided by package io or package os, and those implementations should be preferred in new code. See the specific function documentation for details. (staticcheck)
        "io/ioutil"
        ^
```
